src/routerd_lib/handlers/proxy.cpp

```diff
@@ -45,26 +46,82 @@ namespace NAC {
         return hosts.front();
     }
 
+#ifdef AC_DEBUG_ROUTERD_PROXY
+    void TRouterDProxyHandler::print_outgoing_request(std::shared_ptr<TRouterDRequest> request) const{
+        auto &outgoing_request = request->GetOutGoingRequest();
+        std::cerr << "== OUTGOING REQUEST == " << std::endl;
+        std::cerr << "=== headers ===" << std::endl;
+        for (auto && [header, values] : outgoing_request.Headers()) {
```

Comment:
`for (auto && [header, values]` -> `for (auto&& [header, values]`

And please add empty line before "for".